add note on docker networking (#17306)
PrefectHQ/prefect: docs/v3/manage/self-host.mdx

Whenever that profile is active, the API endpoint is at that address.
See [Profiles and configuration](/v3/develop/settings-and-profiles/) for more information on profiles and configurable Prefect settings.

### Running a server in a Docker container

If you're running a server in a Docker container, you will need to:
- Port forward the server to your local machine (e.g. `-p 4200:4200`)
- Set the `--host` flag on the `prefect server start` command to `0.0.0.0`

For example:

```bash

> docker run -p 4200:4200 --rm -d prefecthq/prefect:3-latest -- prefect server start --host 0.0.0.0

Reviewer: Editorial suggestion: reverse the order of the flags in the command and in the `<Tip>`, since chronologically the container first runs in detached mode before it is stopped and removed.

```suggestion
docker run -p 4200:4200 -d --rm prefecthq/prefect:3-latest -- prefect server start --host 0.0.0.0
```

Reviewer: I'm also happy to be overruled on this if there's a good reason to not swap them.

Author: good call!

```

<Tip>

> The `--rm` flag will remove the container when it is stopped and the `-d` flag will run the container in detached mode (i.e. in the background).

Reviewer: Reverse the order as mentioned above. And I think it's slightly clearer to shift the `-d` part of the sentence from passive to active tense.

```suggestion
The `-d` flag runs the container in detached mode (i.e. in the background) and the `--rm` flag removes the container once it is stopped.
```

Reviewer: I'm also completely happy to be overruled on this if there's a good reason to not swap them.

</Tip>

> Read more about Docker networking [here](https://docs.docker.com/network/network-tutorial-host/).

Reviewer: Avoid the use of 'here' in link text, and instead use the name of the page you're linking to. https://developers.google.com/style/link-text?hl=en#write-link-text

```suggestion
For more information, see the Docker topic about [Networking using the host network](https://docs.docker.com/network/network-tutorial-host/).
```
